R/retrieve_credentials.R

```diff
@@ -1,13 +1,153 @@
+#' We can use `install.packages("keyring")` to install **keyring**.
+#'
+#' @examples
+#' # View the available SMTP credentials
+#' # that are in the system's secure
+#' # key-value store; the `id` values
+#' # in the returned tibble provide what's
+#' # necessary to send email through
+#' # `smtp_send()` and the `creds_key()`
+#' # credential helper function
+#'
+#' # view_credential_keys()
 #'
 #' @export
 view_credential_keys <- function() {
 
+  # Viewing credentials that are on the system-wide key-value
+  # store requires the installation of the keyring package
+  if (!requireNamespace("keyring", quietly = TRUE)) {
+
+    stop("The `keyring` package is required for using the ",
+         "`view_credential_keys()` function",
+         call. = FALSE)
+  }
+
   get_keyring_creds_table()
 }
 
+#' Delete a single **blastula** credential key
+#'
+#' It may be important to delete a credential key and the
+#' `delete_credential_key()` function makes this possible. To understand which
+#' keys are available in the key-value store (and to get their `id` values), use
+#' the [view_credential_keys()] function.
+#'
+#' Support for using the `delete_credential_key()` function (and for doing any
+#' credential key management) is provided through the **keyring** package. This
+#' function cannot be used without that package being available on the system.
+#' We can use `install.packages("keyring")` to install **keyring**.
+#'
+#' @examples
+#' # Delete the credential key with
+#' # the `id` value of `outlook`
+#'
+#' # delete_credential_key("outlook")
+#'
+#' @export
+delete_credential_key <- function(id) {
+
+  # Deleting credentials that are on the system-wide key-value
+  # store requires the installation of the keyring package
+  if (!requireNamespace("keyring", quietly = TRUE)) {
+
+    stop("The `keyring` package is required for using the ",
+         "`delete_credential_key()` function",
+         call. = FALSE)
+  }
+
+  creds_tbl <- get_keyring_creds_table()
+
+  # Get a vector of key names, otherwise known as service names in `keyring`
+  ids_available <- creds_tbl$id
+
+  # Stop if there are no credential keys available
+  if (length(ids_available) < 1) {
+    stop("There are no credential keys available, so there is nothing to delete.",
+         call. = FALSE)
+  }
+
+  # Stop if the provided `id` doesn't match any of those available
+  if (!(id %in% ids_available)) {
+    stop("The specified `id` doesn't correspond to a credential key:\n",
+         "* Use the `view_credential_keys()` function to examine which `id` values are valid",
+         call. = FALSE)
+  }
+
+  # Obtain the matching service name that corresponds to the `id` value
+  key_to_delete <-
+    creds_tbl %>%
+    dplyr::filter(id == {{ id }}) %>%
+    dplyr::pull(key_name)
+
+  # Delete with `keyring::key_delete()`
+  keyring::key_delete(service = key_to_delete)
+
+  message(paste0("The blastula credential key with the `id` of \"", id, "\" has been deleted."))
```

Comment:
Let's not have success messages that don't indicate any useful information other than success.

Reply:
In the two deletion functions, the message has been removed (with `invisible()` used instead).